core/typeSugar.ml

```diff
@@ -1628,18 +1628,41 @@ let rec close_pattern_type : pattern list -> Types.datatype -> Types.datatype =
        (* TODO: expand applications? *)
       | `Application _ -> t
 
-let unify ~pos ~(handle:Gripers.griper) ((_,ltype as t1), (_,rtype as t2)) =
-  try
-    Utils.unify (ltype, rtype)
-  with
-      Unify.Failure error ->
-        begin
-          match error with
-            | `Msg s ->
-                Debug.print ("Unification error: "^s)
-            | _ -> ()
-        end;
-        handle ~pos ~t1 ~t2 ~error
+type unifyResult = UnifySuccess | UnifyFailure of (Unify.error * SourceCode.pos)
+
+let raiseUnify ~(handle:Gripers.griper) ~pos error t1 t2 =
```

Comment:
**Disclaimer** the following is nitpick. Throughout the code base we predominantly use shake case for function names and type names, although, some modules do use camel case. In spirit of converging to a consistent coding style throughout the code base, I suppose `raiseUnify` should be `raise_unify`.

Reply:
I think you mean "snake case". But yes you're right, compound variable names should never use camel case. This is the standard convention for OCaml source code. The conventions for modules and type constructors are a little less clear, but we tend to use camel case for those.

@jstolarek please change the variable and type names to use underscores.

Reply:
@slindley yes, I meant to type "snake case"...

Reply:
> for modules and type constructors (...) we tend to use camel case.
> please change the variable and type names to use underscores.

So, do you want the type constructor `unifyResult` to remain in camel case or do you want it to become `unify_result`?

Reply:
it should be unify_result - as should any other compound name that begins with a lower case letter